app/src/main/java/com/bashkevich/androidfundamentals/model/workmanager/MovieLoaderWorker.kt

```diff
@@ -24,46 +21,53 @@ class MovieLoaderWorker(appContext: Context, params: WorkerParameters) : Corouti
     override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
         try {
             val database = MoviesDatabase.getDatabase(applicationContext)
-            var nowPlayingMoviesIds = listOf<Int>()
-            var popularMoviesIds = listOf<Int>()
-            var topRatedMoviesIds = listOf<Int>()
-            var upcomingMoviesIds = listOf<Int>()
-            val allMoviesIds: List<Int>
+            val sharedPreferences =
+                androidx.preference.PreferenceManager.getDefaultSharedPreferences(applicationContext)
+            var nowPlayingMoviesIds: Deferred<List<Int>>
+            var popularMoviesIds: Deferred<List<Int>>
+            var topRatedMoviesIds: Deferred<List<Int>>
+            var upcomingMoviesIds: Deferred<List<Int>>
+            val moviesFromNetworkIds: List<Int>
+
+            val prevMovies = database.moviesDao.getAllMovies()
+
+            if (sharedPreferences.getInt(NEW_MOVIE_ID, 0) != 0) {
+                val editor = sharedPreferences.edit()
+                editor.remove(NEW_MOVIE_ID)
+                editor.apply()
+            }
 
             coroutineScope {
                 launch {
+                    database.moviesDao.deleteAllGenres()
```

Comment:
Why delete all actors and genres?

Reply:
In order not to lose genre/actor with the film tgat is absent, so store this genre/actor will be useless